### PR TITLE
#19 Waiting for the environment for Create and Update

### PR DESCRIPTION
**Pull request by ja6a**
1. Waiting until all the environment's VMs are started on creation. 
2. Waiting for runstate to not be busy when updating.

### CODE REVIEW

File: skytap/resource_skytap_environment.go [hunk 1]
```diff
@@ -144,6 +147,21 @@ func resourceSkytapEnvironmentCreate(d *schema.ResourceData, meta interface{}) e
 
 	d.SetId(*environment.ID)
 
+	stateConf := &resource.StateChangeConf{
+		Pending:    resourceSkytapEnvironmentPendingCreateRunstates,
+		Target:     resourceSkytapEnvironmentTargetCreateRunstates,
+		Refresh:    resourceSkytapEnvironmentCreateRunstateRefreshFunc(d, meta),
```
Comment (pegerto) [inline skytap/resource_skytap_environment.go]: [minor] we remove resourceSkytap from the variable name  / function for readability

File: skytap/resource_skytap_environment.go [hunk 2]
```diff
@@ -246,3 +279,85 @@ func resourceSkytapEnvironmentDelete(d *schema.ResourceData, meta interface{}) e
...
+
+		log.Printf("[INFO] retrieving environment: %s", id)
+		environment, err := client.Get(ctx, id)
+
+		if err != nil {
+			log.Printf("[WARN] Error on retrieving environment status (%s) when waiting: %s", id, err)
+			return nil, "", err
+		}
+
+		var runstateCount int
```
Comment (pegerto) [inline skytap/resource_skytap_environment.go]: this logic can be easily refactor
```
state = running
for each vm:
     if vm.state != running || vm.state is error:
              state = vm.state 
               break

 
```

File: skytap/resource_skytap_environment.go (hunk 1 above)
Comment (ja6a) [reply skytap/resource_skytap_environment.go]: done

File: skytap/resource_skytap_environment.go (hunk 2 above)
Comment (ja6a) [reply skytap/resource_skytap_environment.go]: done

**Pull request merged**